src/util/metadata/global.rs

```diff
@@ -13,16 +13,23 @@ use super::header_metadata::HeaderMetadataSpec;
 /// Each plan or policy which uses a metadata bit-set, needs to create an instance of this struct.
 ///
 /// For performance reasons, objects of this struct should be constants.
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Copy, Debug)]
 pub enum MetadataSpec {
     InHeader(HeaderMetadataSpec),
     OnSide(SideMetadataSpec),
 }
 
 impl MetadataSpec {
-    pub fn is_on_side(&self) -> bool {
+    pub const fn is_on_side(&self) -> bool {
         matches!(self, &MetadataSpec::OnSide(_))
     }
+
+    pub const fn extract_side_spec(&self) -> &SideMetadataSpec {
```

Comment:
As this is a public method, probably a short comment is useful

Reply:
Sure.